--- rpm-4.4.2/Doxyfile.in.doxy	2006-04-12 14:36:03.000000000 -0400
+++ rpm-4.4.2/Doxyfile.in	2006-04-12 14:36:15.000000000 -0400
@@ -599,7 +599,6 @@
                          @top_srcdir@/python/rpmte-py.h \
                          @top_srcdir@/python/rpmts-py.c \
                          @top_srcdir@/python/rpmts-py.h \
-                         @top_srcdir@/tools/dump.c \
                          @top_srcdir@/tools/dumpdb.c \
                          @top_srcdir@/tools/javadeps.c \
                          @top_srcdir@/tools/rpmcache.c \
