--- rpm-4.4.1/lib/rpmlib.h.prereq	2005-03-31 18:23:26.175045641 +0200
+++ rpm-4.4.1/lib/rpmlib.h	2005-03-31 18:24:41.549762818 +0200
@@ -501,8 +501,7 @@
     RPMSENSE_EQUAL	= (1 << 3),
     RPMSENSE_PROVIDES	= (1 << 4), /* only used internally by builds */
     RPMSENSE_CONFLICTS	= (1 << 5), /* only used internally by builds */
-	/* bit 6 used to be RPMSENSE_PREREQ */
-#define	RPMSENSE_PREREQ	RPMSENSE_ANY
+    RPMSENSE_PREREQ     = (1 << 6),     /*!< @todo Legacy. */
     RPMSENSE_OBSOLETES	= (1 << 7), /* only used internally by builds */
     RPMSENSE_INTERP	= (1 << 8),	/*!< Interpreter used by scriptlet. */
     RPMSENSE_SCRIPT_PRE	= ((1 << 9)|RPMSENSE_PREREQ), /*!< %pre dependency. */
