--- rpm-4.4.2/build/files.c.excludedsize	2005-11-29 16:21:12.000000000 -0500
+++ rpm-4.4.2/build/files.c	2005-11-29 16:21:21.000000000 -0500
@@ -1678,7 +1678,7 @@
 	} else
 	    i = fl->fileListRecsUsed;
 
-	if (S_ISREG(flp->fl_mode) && i >= fl->fileListRecsUsed)
+	if (!(flp->flags & RPMFILE_EXCLUDE) && S_ISREG(flp->fl_mode) && i >= fl->fileListRecsUsed) 
 	    fl->totalFileSize += flp->fl_size;
     }
 
