--- rpm-4.4.1/build/rpmfc.c.noconfig	2005-04-26 11:59:59.000000000 +0200
+++ rpm-4.4.1/build/rpmfc.c	2005-04-26 12:02:31.000000000 +0200
@@ -1511,11 +1511,13 @@
     fi = rpmfiInit(fi, 0);
     if (fi != NULL)
     while ((c = rpmfiNext(fi)) >= 0) {
+#ifdef RPM_WITH_CONFIG_DEPENDENCY
 	rpmfileAttrs fileAttrs;
 
 	/* Does package have any %config files? */
 	fileAttrs = rpmfiFFlags(fi);
 	genConfigDeps |= (fileAttrs & RPMFILE_CONFIG);
+#endif
 
 	av[c] = xstrdup(rpmfiFN(fi));
     }
@@ -1537,6 +1539,7 @@
 	xx = headerRemoveEntry(pkg->header, RPMTAG_PROVIDEVERSION);
 	xx = headerRemoveEntry(pkg->header, RPMTAG_PROVIDEFLAGS);
 
+#ifdef RPM_WITH_CONFIG_DEPENDENCY
 	/* Add config dependency, Provides: config(N) = EVR */
 	if (genConfigDeps) {
 	    N = rpmdsN(pkg->ds);
@@ -1549,6 +1552,7 @@
 	    xx = rpmdsMerge(&fc->provides, ds);
 	    ds = rpmdsFree(ds);
 	}
+#endif
     }
 
     if (!fc->skipReq) {
@@ -1559,6 +1563,7 @@
 	xx = headerRemoveEntry(pkg->header, RPMTAG_REQUIREVERSION);
 	xx = headerRemoveEntry(pkg->header, RPMTAG_REQUIREFLAGS);
 
+#ifdef RPM_WITH_CONFIG_DEPENDENCY
 	/* Add config dependency,  Requires: config(N) = EVR */
 	if (genConfigDeps) {
 	    N = rpmdsN(pkg->ds);
@@ -1571,6 +1576,7 @@
 	    xx = rpmdsMerge(&fc->requires, ds);
 	    ds = rpmdsFree(ds);
 	}
+#endif
     }
 
     /* Build file class dictionary. */
