diff -up rpm-4.4.2.1-rc3/configure.ac.orig rpm-4.4.2.1-rc3/configure.ac
--- rpm-4.4.2.1-rc3/configure.ac.orig	2007-07-21 15:11:25.000000000 -0400
+++ rpm-4.4.2.1-rc3/configure.ac	2007-07-21 15:12:04.000000000 -0400
@@ -1132,11 +1132,13 @@ fi
 #
 # get rid of the 4-th tuple, if config.guess returned "linux-gnu" for build_os
 #
+build_os_gnu=-gnu
 if echo "$build_os" | grep '.*-gnulibc1' > /dev/null ; then
 	build_os=`echo "${build_os}" | sed 's/-gnulibc1$//'`
 fi
 if echo "$build_os" | grep '.*-gnueabi' > /dev/null ; then
 	build_os=`echo "${build_os}" | sed 's/-gnueabi$//'`
+	build_os_gnu=-gnueabi
 fi
 if echo "$build_os" | grep '.*-gnu' > /dev/null ; then
 	build_os=`echo "${build_os}" | sed 's/-gnu$//'`
@@ -1231,12 +1233,14 @@ unknown|pc|ibm|redhat|pld|mandrake|conec
 	;;
 esac
 RPMCANONOS="$build_os_noversion"
+RPMCANONGNU="$build_os_gnu"
 AC_SUBST(RPMCANONCOLOR)
 AC_SUBST(autorelocate_path)
 AC_SUBST(autorelocate_dcolor)
 AC_SUBST(RPMCANONARCH)
 AC_SUBST(RPMCANONVENDOR)
 AC_SUBST(RPMCANONOS)
+AC_SUBST(RPMCANONGNU)
 
 if test X"$prefix" = XNONE ; then
     usrprefix="$ac_default_prefix"
diff -up rpm-4.4.2.1-rc3/macros.in.orig rpm-4.4.2.1-rc3/macros.in
--- rpm-4.4.2.1-rc3/macros.in.orig	2007-07-21 15:12:13.000000000 -0400
+++ rpm-4.4.2.1-rc3/macros.in	2007-07-21 15:12:32.000000000 -0400
@@ -819,7 +819,7 @@ print (t)\
 %_build_arch		@RPMCANONARCH@
 %_vendor		@RPMCANONVENDOR@
 %_os			@RPMCANONOS@
-%_gnu			-gnu
+%_gnu			@RPMCANONGNU@
 %_target_platform	%{_target_cpu}-%{_vendor}-%{_target_os}%{?_gnu}
 
 #
