--- rpm-4.4.2/macros.in.perlreq	2005-07-26 12:42:38.000000000 -0400
+++ rpm-4.4.2/macros.in	2005-07-26 12:42:47.000000000 -0400
@@ -430,10 +430,10 @@
 #
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by @RPMCONFIGDIR@/rpmdeps {--provides|--requires}.
-%__perl_provides	@RPMCONFIGDIR@/perldeps.pl --provides
-%__perl_requires	@RPMCONFIGDIR@/perldeps.pl --requires
-#%__perl_provides	@RPMCONFIGDIR@/perl.prov
-#%__perl_requires	@RPMCONFIGDIR@/perl.req
+#%__perl_provides	@RPMCONFIGDIR@/perldeps.pl --provides
+#%__perl_requires	@RPMCONFIGDIR@/perldeps.pl --requires
+%__perl_provides	@RPMCONFIGDIR@/perl.prov
+%__perl_requires	@RPMCONFIGDIR@/perl.req
 
 %__python_provides	@RPMCONFIGDIR@/pythondeps.sh --provides
 %__python_requires	@RPMCONFIGDIR@/pythondeps.sh --requires
