--- rpm-4.4.2/gendiff.cdiff	2002-07-13 20:54:36.000000000 +0100
+++ rpm-4.4.2/gendiff	2007-04-16 12:48:09.000000000 +0100
@@ -6,14 +6,16 @@
   exit 1
 }
 
+: ${DIFF:=diff -p}
+
 find $1 \( -name "*$2" -o -name ".*$2" \) -print |
 while read f; do
     U=-u
     [ "`basename $f`" = "ChangeLog$2" ] && U=-U0
-#    diff ${U} $f `echo $f | sed s/$2\$//`
+#    ${DIFF} ${U} $f `echo $f | sed s/$2\$//`
     if [ -r "$f" ]; then
-	diff ${U} "${f}" "${f%$2}"
+	${DIFF} ${U} "${f}" "${f%$2}"
     else
-	diff ${U} /dev/null "${f%$2}"
+	${DIFF} ${U} /dev/null "${f%$2}"
     fi
 done
