--- rpm-4.4.2/macros.in.mono	2005-11-15 17:45:13.000000000 -0500
+++ rpm-4.4.2/macros.in	2005-11-15 17:45:44.000000000 -0500
@@ -438,6 +438,9 @@
 %__python_provides	@RPMCONFIGDIR@/pythondeps.sh --provides
 %__python_requires	@RPMCONFIGDIR@/pythondeps.sh --requires
 
+%__mono_provides        @RPMCONFIGDIR@/mono-find-provides %{_builddir}/%{?buildsubdir} %{buildroot} %{_libdir}
+%__mono_requires        @RPMCONFIGDIR@/mono-find-requires %{_builddir}/%{?buildsubdir} %{buildroot} %{_libdir}
+
 #
 # fixowner, fixgroup, and fixperms are run at the end of hardcoded setup
 # These macros are necessary only for legacy compatibility, and have moved
--- rpm-4.4.2/build/rpmfc.h.mono	2005-11-15 17:09:26.000000000 -0500
+++ rpm-4.4.2/build/rpmfc.h	2005-11-15 17:10:04.000000000 -0500
@@ -77,6 +77,7 @@
     RPMFC_PYTHON		= (1 << 26),
     RPMFC_PHP			= (1 << 27),
     RPMFC_TCL			= (1 << 28),
+    RPMFC_MONO                  = (1 << 6),
 
     RPMFC_WHITE			= (1 << 29),
     RPMFC_INCLUDE		= (1 << 30),
--- rpm-4.4.2/build/rpmfc.c.mono	2005-11-15 16:04:46.000000000 -0500
+++ rpm-4.4.2/build/rpmfc.c	2005-11-15 17:07:20.000000000 -0500
@@ -458,6 +458,10 @@
   /* XXX "python 2.3 byte-compiled" */
   { "python ",			RPMFC_PYTHON|RPMFC_INCLUDE },
 
+  /* XXX .NET executables and libraries.  file(1) cannot differ from win32 
+   * executables unfortunately :( */
+  { "PE executable",            RPMFC_MONO|RPMFC_INCLUDE },
+
   { "current ar archive",	RPMFC_STATIC|RPMFC_LIBRARY|RPMFC_ARCHIVE|RPMFC_INCLUDE },
 
   { "Zip archive data",		RPMFC_COMPRESSED|RPMFC_ARCHIVE|RPMFC_INCLUDE },
@@ -731,6 +735,11 @@
 #endif
 	    xx = rpmfcHelper(fc, 'R', "python");
     }
+    if (fc->fcolor->vals[fc->ix] & RPMFC_MONO) {
+        xx = rpmfcHelper(fc, 'P', "mono");
+        if (is_executable)
+            xx = rpmfcHelper(fc, 'R', "mono");
+    }
 
     return 0;
 }
@@ -1059,6 +1068,7 @@
     { rpmfcELF,		RPMFC_ELF },
     { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_PERL) },
     { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_PYTHON) },
+    { rpmfcSCRIPT,      RPMFC_MONO },
     { NULL, 0 }
 };
 
