diff -up rpm-4.4.2.1-rc1/popt/configure.poptver rpm-4.4.2.1-rc1/popt/configure
--- rpm-4.4.2.1-rc1/popt/configure.poptver	2007-06-26 13:51:35.000000000 +0300
+++ rpm-4.4.2.1-rc1/popt/configure	2007-06-26 13:51:45.000000000 +0300
@@ -2418,7 +2418,7 @@ fi
 
 # Define the identity of the package.
  PACKAGE=popt
- VERSION=1.10.2
+ VERSION=1.10.2.1
 
 
 cat >>confdefs.h <<_ACEOF
diff -up rpm-4.4.2.1-rc1/popt/configure.ac.poptver rpm-4.4.2.1-rc1/popt/configure.ac
--- rpm-4.4.2.1-rc1/popt/configure.ac.poptver	2007-06-26 13:51:29.000000000 +0300
+++ rpm-4.4.2.1-rc1/popt/configure.ac	2007-06-26 13:51:40.000000000 +0300
@@ -2,7 +2,7 @@ AC_INIT(popt.h)
 AC_CANONICAL_SYSTEM
 AC_PREREQ(2.12)
 AC_CONFIG_HEADERS([config.h])
-AM_INIT_AUTOMAKE(popt, 1.10.2)
+AM_INIT_AUTOMAKE(popt, 1.10.2.1)
 
 ALL_LINGUAS="cs da de es eu_ES fi fr gl hu id is it ja ko no pl pt pt_BR ro ru sk sl sr sv tr uk wa zh zh_CN zh_TW"
 
