--- rpm-4.8.0/scripts/postscriptdriver.prov	2010-02-05 15:40:34.451050057 +0000
+++ rpm-4.8.0/scripts/postscriptdriver.prov~	2010-02-05 15:44:10.141926520 +0000
@@ -196,9 +196,15 @@ class DrvDriver(PPDDriver):
                  self.drv ]
 
         signal.alarm (60)
-        p = subprocess.Popen (argv,
-                              stdout=subprocess.PIPE,
-                              stderr=subprocess.PIPE)
+        try:
+            p = subprocess.Popen (argv,
+                                  stdout=subprocess.PIPE,
+                                  stderr=subprocess.PIPE)
+        except OSError:
+            # ppdc not available.
+            os.rmdir (outputdir)
+            return Driver.list (self)
+
         try:
             (stdout, stderr) = p.communicate ()
             signal.alarm (0)
@@ -230,7 +236,7 @@ class TagBuilder:
                     self.ids += DynamicDriver (path).list ()
                 except TimedOut:
                     pass
-                except OSError, e:
+                except OSError, (e, s):
                     if e == errno.EACCES or e == errno.ENOENT:
                         # Not executable
                         pass
