--- rpm-4.4.1/macros.in.fedora	2005-03-07 11:37:37.000000000 -0500
+++ rpm-4.4.1/macros.in	2005-03-07 11:37:48.000000000 -0500
@@ -658,8 +658,8 @@
 
 # Horowitz Key Protocol server configuration
 #
-%_hkp_keyserver         hkp://pgp.mit.edu
-%_hkp_keyserver_query   %{_hkp_keyserver}/pks/lookup?op=get&search=0x
+#%_hkp_keyserver         hkp://pgp.mit.edu
+#%_hkp_keyserver_query   %{_hkp_keyserver}/pks/lookup?op=get&search=0x
 
 #==============================================================================
 # ---- Transaction macros.
