commit 3b9636cd69b81b6e23331bb797c0fdde79e97316
Author: Jindrich Novy <jnovy@redhat.com>
Date:   Thu Aug 14 13:08:09 2008 +0200

    Accept numerical characters in macro expansion (related to rhbz#456103)

diff --git a/build/expression.c b/build/expression.c
index 970ec05..83a4021 100644
--- a/build/expression.c
+++ b/build/expression.c
@@ -272,7 +272,7 @@ static int rdToken(ParseState state)
       char *temp;
       size_t ts;
 
-      for (ts=1; p[ts] && (risalpha(p[ts]) || p[ts] == '_'); ts++);
+      for (ts=1; p[ts] && (risalnum(p[ts]) || p[ts] == '_'); ts++);
       temp = xmalloc(ts+1);
       memcpy(temp, p, ts);
       p += ts-1;
