--- rpm-4.4.2/lib/transaction.c.netsharedpath	2006-02-13 18:15:37.000000000 -0500
+++ rpm-4.4.2/lib/transaction.c	2006-02-13 18:15:38.000000000 -0500
@@ -820,6 +820,9 @@
 		    /*@innercontinue@*/ continue;
 		if (strncmp(dn, *nsp, dnlen))
 		    /*@innercontinue@*/ continue;
+		/* Insure that only the netsharedpath basename is compared. */
+		if ((s = strchr((*nsp) + dnlen, '/')) != NULL && s[1] != '\0')
+		    /*@innercontinue@*/ continue;
 		if (strncmp(bn, (*nsp) + dnlen, bnlen))
 		    /*@innercontinue@*/ continue;
 		len = dnlen + bnlen;
