commit 6eba5dc468aa2d26ff80f9321e8d4d014279d506
Author: Panu Matilainen <pmatilai@redhat.com>
Date:   Tue Mar 22 08:36:35 2011 +0200

    Fix classification of ELF binaries with sticky bit (RhBug:689182)

diff --git a/fileattrs/elf.attr b/fileattrs/elf.attr
index fba180b..bc6ce83 100644
--- a/fileattrs/elf.attr
+++ b/fileattrs/elf.attr
@@ -1,4 +1,4 @@
 %__elf_provides		%{_rpmconfigdir}/elfdeps --provides %{?__filter_GLIBC_PRIVATE:--filter-private}
 %__elf_requires		%{_rpmconfigdir}/elfdeps --requires %{?__filter_GLIBC_PRIVATE:--filter-private}
-%__elf_magic		^ELF (32|64)-bit.*$
+%__elf_magic		^(sticky )?ELF (32|64)-bit.*$
 %__elf_flags		exeonly
